### DISCUSSION TITLE
How are virtual parts assigned to sales orders?

### DISCUSSION THREAD

**Discussion by miggland** [inventree/InvenTree#3262] [Q&A] [answered]
Hi,

I have added a virtual part to a sales order. It's the only part in the sales order. So far so good.

Before I can complete the order, I need to assign stock - I have no stock yet, so I go the part page ( ... /part/107/), Stock page and click "New Stock item":

![image](https://user-images.githubusercontent.com/1314853/175980441-ec425316-69b0-4723-b569-eece3ea2ee5b.png)

However, when I try to submit I get the following:

![image](https://user-images.githubusercontent.com/1314853/175980504-cdb28c1c-5beb-4271-af1a-24f27d42da0b.png)

So I'm stuck - I have no stock to assign in the sales order, and I can't assign any because of that error message.

Is this a bug, or is this not the intended procedure?

**Comment by matmair** [comment, marked as answer]
@miggland this seems to be a bug. Please file a bug report.
